Transform assays

myushen opened the pull request:
This PR includes updates to the following items:
- Updated function and module to handle Anndata input.
- Option for gene nomenclature.
- Module for identifying empty droplets based on nFeature_RNA and nCount_RNA thresholds.
- Ensuring the transformed assay is saved in the same format as the input.

myushen committed:
update functions for anndata input work

myushen committed:
update module for anndata/sce input

myushen committed:
merge, solve conflicts from master

Merge branch 'master' into module-data-input

Conflicts:
	R/functions.R
	R/modules_grammar_hpc.R

myushen committed:
fix

myushen committed:
fix package dependency

myushen committed:
fix mt-ribsome gene identification and pipeline running

myushen committed:
fix

myushen committed:
merge from master

Merge branch 'master' into module-data-input

Conflicts:
	tests/testthat/test_single_functions.R

myushen committed:
fix conflicts

Merge remote-tracking branch 'remotes/upstream/custom_input_names' into transform-assays

Conflicts:
	R/functions.R
	R/modules_grammar_hpc.R
	tests/testthat/test_single_functions.R

myushen committed:
save transform assay in different formats

myushen committed:
fix pipeline

myushen committed:
fix the pipeline for census anndata input

myushen committed:
testthat for census data

myushen committed:
update transformation function

stemangiola committed:
add API constructor for user to add custom functions

myushen committed:
update transformation function

myushen committed:
empty_droplet based on custom filter threshold

myushen committed:
Merge branch 'custom_input_names' into transform-assays

stemangiola committed:
fix quote passing

stemangiola commented on R/functions.R:
```diff
@@ -62,22 +63,28 @@ annotation_label_transfer <- function(input_read_RNA_assay,
+  } else if (gene_nomenclature == "symbol") {
+    blueprint <- celldex::BlueprintEncodeData(legacy = TRUE)
+    
+  }
```
A note on this code. It can be simplified as

```r
blueprint <- celldex::BlueprintEncodeData(
ensembl = gene_nomenclature == "ensemble", 
legacy = TRUE
)
```

Similar cases below

stemangiola commented on R/functions.R:
```diff
@@ -314,17 +327,17 @@ annotation_label_transfer <- function(input_read_RNA_assay,
                                  annotation_column = NULL,
-                                 assay = NULL) {
+                                 assay = NULL,
+                                 gene_nomenclature) {
```
`gene_nomenclature` is a target in this case (I don't remember if I told you to put it as a dynamic argument, apologies for that in case).

It is better as a target because it is used many many times within the pipeline.

myushen replied:
did you mean `gene_nomenclature` need to be tiered, or it needs to be put as a module argument?

stemangiola replied:
gene_nomenclature should be a target that is not tiered, is just N=1

myushen replied:
yeah I have defined it as a target in the main function `initialise_hpc` unless I misunderstand your point.

stemangiola replied:
if you defined it in `initialise_hpc` there is no need to define it again

myushen replied:
If the `gene_nomenclature` parameter is absent in the function, there are certain operations in the function that need minor tweak based on `nomenclature`, like the cc.gene transformation. How does the target module decide which `nomenclature` to use?

stemangiola commented on R/functions.R:
```diff
@@ -545,7 +563,7 @@ alive_identification <- function(input_read_RNA_assay,
 doublet_identification <- function(input_read_RNA_assay, 
                                    empty_droplets_tbl, 
                                    alive_identification_tbl, 
-                                   #annotation_label_transfer_tbl, 
```
I remember I commented out this argument because it was not use anymore, as tablet identification used internal plastering rather than external annotation (not worth in my opinion).

stemangiola commented on R/modules_grammar_hpc.R:
```diff
@@ -168,12 +169,12 @@ remove_empty_DropletUtils.Seurat = function(input_hpc, total_RNA_count_check = N
 
 #' @export
-remove_empty_DropletUtils.HPCell = function(input_hpc, total_RNA_count_check = NULL, target_input = "read_file", target_output = "empty_tbl",...) {
+remove_empty_threshold.HPCell = function(input_hpc, RNA_count_threshold = NULL, RNA_feature_threshold = NULL, target_input = "read_file", target_output = "empty_tbl",...) {
```
I think, if you invert the order of `remove_empty_DropletUtils` first, and `remove_empty_threshold` after, Github would not get confused thinking you have modified the original function.

It is a small detail but quite confusing for code review.

stemangiola commented on R/modules_grammar_hpc.R:
```diff
@@ -274,20 +373,21 @@ remove_dead_scuttle.HPCell = function(input_hpc, group_by = NULL, target_input =
             annotation_tbl,
-            grouping_column
+            grouping_column,
+            gene_nomenclature = gene_nomenclature
```
Same as above, she nomenclature should be dropped from all modules, and use internally the `gene_nomenclature ` target

stemangiola commented on R/modules_grammar_hpc.R:
```diff
@@ -861,78 +966,78 @@ get_single_cell.HPCell = function(input_hpc, factors_to_regress = NULL, target_i
 #' @param .contrasts Contrasts parameter.
 #' @return The result of the differential abundance test.
 #'
-setMethod(
```
There is no need to comment this out, it is just a method and you can choose to not use it. Again, if you commented you're creating a lot of noise for that things you have changed a lot of stuff, making it a bit difficult for code reviewers.

stemangiola reviewed (commented):
Very good PR. Please see my comments, for some changes.

stemangiola committed:
fix framework, now it works except it does not know for which arguments it should tier and which are of dimension 1

stemangiola committed:
some fixes

stemangiola committed:
fix the hpc_merge

stemangiola committed:
include pseudobulk and get_single_cell

stemangiola committed:
fix DE

stemangiola committed:
refactor code eliminating one function

stemangiola committed:
update readme

stemangiola committed:
replace default input target name, and replace undefined functions

stemangiola committed:
add constructor to README

stemangiola committed:
simplified if there is not tiering

stemangiola committed:
update README

myushen committed:
update azimuth reference

stemangiola committed:
Update README.md

myushen committed:
Merge branch 'azimuth_annotation' into transform-assays

myushen committed:
test pilot data

myushen committed:
solve conflicts

Merge branch 'transform-assays' of github.com:myushen/HPCell into transform-assays

# Conflicts:
#	R/modules_grammar_hpc.R
#	R/utilities.R
#	man/empty_droplet_threshold.Rd
#	tests/testthat/test-census-samples.R

myushen committed:
update azimuth reference

stemangiola committed:
typo README

stemangiola committed:
add reports

stemangiola committed:
update README

stemangiola committed:
Merge branch 'constructor-API' of https://github.com/MangiolaLaboratory/HPCell into constructor-API

stemangiola committed:
Merge pull request #78 from MangiolaLaboratory/constructor-API

Constructor api

stemangiola committed:
Update DESCRIPTION

stemangiola committed:
Update DESCRIPTION

myushen committed:
update azimuth

stemangiola committed:
track files

stemangiola committed:
version UP

myushen committed:
track files and solve conflicts.

Merge branch 'master' into transform-assays

# Conflicts:
#	R/factories.R
#	R/functions.R
#	R/modules_grammar_hpc.R
#	R/tranform_assay.R
#	R/utilities.R
#	man/test_differential_abundance.Rd
#	man/transform_utility.Rd
#	tests/testthat/test_single_functions.R

stemangiola committed:
fix transform assay

stemangiola committed:
add source argument to sutom functions

stemangiola committed:
deal with undefined targets

stemangiola committed:
deal with undefined targets

stemangiola committed:
fixed bug

stemangiola committed:
fixed bug

stemangiola committed:
still a bug with SCT filtering

stemangiola committed:
still a bug with SCT filtering

stemangiola committed:
clean backend _target.R file

myushen committed:
fix transform_assay

myushen committed:
update testthat

myushen committed:
fix fetching blueprint reference

myushen committed:
update testthat

stemangiola committed:
update consensus function

stemangiola committed:
complete the consensus function

stemangiola committed:
improve the consensus function

stemangiola committed:
move cellChat to dev because it is using the version v1 rather than v2, we need to update

stemangiola committed:
drop cellchat dependencies

stemangiola committed:
fix consensus function

stemangiola committed:
update exports

stemangiola committed:
pull

Merge branch 'master' of github.com:Melbourne-COVID-Predict/jascap

# Conflicts:
#	NAMESPACE

stemangiola committed:
mainly update cell consensus

stemangiola committed:
Merge branch 'drop-undefined-datasets' of https://github.com/MangiolaLaboratory/HPCell into drop-undefined-datasets

myushen committed:
update pipeline

stemangiola committed:
Merge pull request #81 from MangiolaLaboratory/drop-undefined-datasets

Drop undefined datasets

myushen committed:
merge master and solve conflicts

Merge remote-tracking branch 'origin/master' into transform-assays

# Conflicts:
#	NAMESPACE
#	R/CellChat.R
#	R/functions.R
#	R/modules_grammar_hpc.R
#	R/tranform_assay.R
#	R/utilities.R
#	tests/testthat/test_single_functions.R
